domain/schema/testing/controllersuite.go

```diff
@@ -48,3 +50,15 @@ func (s *ControllerSuite) ApplyDDLForRunner(c *gc.C, runner coredatabase.TxnRunn
 func (s *ControllerSuite) ControllerTxnRunner() coredatabase.TxnRunner {
 	return s.TxnRunner()
 }
+
+// SeedControllerUUID sets the uuid in the controller table to the default
+// testing value. It does not add any other controller config.
+func (s *ControllerSuite) SeedControllerUUID(c *gc.C) string {
```

Comment:
Why not update the suite struct:
```
type ControllerSuite struct {
	testing.DqliteSuite

        ControllerUUID string
}
```
and call SeedControllerUUID() in ControllerSuite.SetupTest(). Less to remember when cut and pasteing.

Reply:
Because when the ControllerSuite is used within the APISuite it calls the bootstrap function on ControllerConfig in its SetUpTest. This also inserts the controllerUUID, if its already there from this function you get back an error.